src/main/groovy/life/qbic/DownloadRequest.groovy

```diff
@@ -34,6 +37,24 @@ class DownloadRequest {
         dataSetFiles.each { dsFile ->
             addDataSet(dsFile.getPermId().toString(), dsFile)
         }
+        this.retries = 1
+    }
+
+    /**
+     * Download request constructor with a provided list of data set files and a configured
+     * number of retries on failure.
+     *
+     * @param dataSetFiles
+     * @param sampleCode
+     * @param numberRetries The number of retries. Must be >=1, else it will be set to 1
+     */
+    DownloadRequest(List<DataSetFile> dataSetFiles, String sampleCode, int numberRetries) {
+        this()
+        this.sampleCode = sampleCode
+        dataSetFiles.each { dsFile ->
+            addDataSet(dsFile.getPermId().toString(), dsFile)
+        }
+        this.retries = numberRetries
```

Comment:
how do you check it the number of retries is bigger than 1?

Reply:
I would expect a check before setting the number of retries

Reply:
very well spotted!